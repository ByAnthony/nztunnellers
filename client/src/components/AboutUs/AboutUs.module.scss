@use 'src/index.scss';

.container {
  @include index.component-margin;

  margin-top: index.$l-margin;
  margin-bottom: 40px;

  .header {
    margin-bottom: index.$l-margin;

    @include index.tablet {
      margin-top: index.$xl-margin;
    }

    @include index.desktop {
      margin-bottom: 125px;
    }
  }

  a {
    @include index.text-link;

    font-size: 1.15rem;
    font-weight: 500;
    vertical-align: baseline;
    line-height: 100%;

    @include index.desktop {
      font-size: 1.25rem;
      line-height: 2.25rem;
    }
  }

  @include index.tablet {
    margin-top: index.$xl-margin;
  }

  @include index.desktop {
    margin-top: index.$xxl-margin;
  }
}

.image-container {
  display: flex;
  margin: index.$base-margin 0;
  max-width: index.$inner-width;
  justify-content: center;
  align-items: center;

  @include index.tablet {
    margin: index.$l-margin 0;
  }

  @include index.desktop {
    margin: 0 auto;
    margin-top: index.$xl-margin;
    margin-bottom: index.$xl-margin;
  }
}

.image {
  object-fit: cover;
  object-position: center;
  height: 460px;
  width: 100%;
  border-radius: index.$border-radius;

  @include index.tablet {
    height: 515px;
  }

  @include index.desktop {
    height: 575px;
    max-width: index.$inner-width;
  }
}
