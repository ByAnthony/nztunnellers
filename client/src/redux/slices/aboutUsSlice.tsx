import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { AboutUs } from '../../types/about-us';

export const aboutUsApi = createApi({
  reducerPath: 'aboutUsApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getAboutUs: builder.query<AboutUs, void>({
      query: () => ({
        url: '/about-us',
      }),
    }),
  }),
});

export const {
  useGetAboutUsQuery,
} = aboutUsApi;
