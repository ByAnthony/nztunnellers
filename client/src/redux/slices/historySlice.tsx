import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Article, Articles } from '../../types/article';

export const historyApi = createApi({
  reducerPath: 'historyApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getAllHistoryArticleLink: builder.query<Array<Articles>, void>({
      query: () => ({
        url: '/',
      }),
    }),
    getHistoryArticleById: builder.query<Article, string>({
      query: (id: string) => `history/${id}`,
    }),
  }),
});

export const {
  useGetAllHistoryArticleLinkQuery,
  useGetHistoryArticleByIdQuery,
} = historyApi;
