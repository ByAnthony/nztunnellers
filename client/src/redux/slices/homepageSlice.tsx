import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Articles } from '../../types/article';

export const homepageApi = createApi({
  reducerPath: 'homepageApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getHomepageData: builder.query<Record<string, Array<Articles>>, void>({
      query: () => ({
        url: '/',
      }),
    }),
  }),
});

export const {
  useGetHomepageDataQuery,
} = homepageApi;
