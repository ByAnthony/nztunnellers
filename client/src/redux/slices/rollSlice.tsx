import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Details } from '../../types/roll';
import { Profile } from '../../types/tunneller';

export const tunnellersApi = createApi({
  reducerPath: 'tunnellersApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'http://localhost:5000/',
  }),
  endpoints: (builder) => ({
    getAllTunnellers: builder.query<Record<string, Array<Details>>, void>({
      query: () => ({
        url: 'roll/',
      }),
    }),
    getTunnellerById: builder.query<Profile, number>({
      query: (id: number) => `roll/${id}`,
    }),
  }),
});

export const { useGetAllTunnellersQuery, useGetTunnellerByIdQuery } = tunnellersApi;
