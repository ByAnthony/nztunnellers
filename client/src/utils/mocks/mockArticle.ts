import { AboutUs } from '../../types/about-us';
import {
  Article, Image, Next, Section,
} from '../../types/article';

export const mockTitle: string = 'My Awesome\\Article Title';

export const mockChapter: number = 1;

export const mockTopImage: Image = {
  file: 'img-123.png',
  alt: 'Accessible alt text',
};

export const mockArticleImage: Image = {
  ...mockTopImage,
  title: 'An Amazing Photo',
  photographer: 'The French Photographer Doisneau',
  reference: 'Where the photo is preserved',
};

export const mockSectionOne: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockSectionTwo: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockNextButton: Next = {
  url: 'my-path-to-next-chapter/',
  title: 'Next\\Chapter',
  chapter: 3,
};

export const mockNotes: string = '[1.](#reference_1) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\n[2.](#reference_2) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

export const mockArticle: Article = {
  id: 'my-awesome-article',
  chapter: mockChapter,
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage, mockArticleImage],
  next: mockNextButton,
  notes: mockNotes,
};

export const mockAboutUs: AboutUs = {
  id: 'my-awesome-article',
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
