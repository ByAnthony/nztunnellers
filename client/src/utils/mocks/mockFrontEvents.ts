import { Date, Event, EventDetail } from '../../types/tunneller';

const mockDate = (year: string | null, dayMonth: string | null): Date => ({
  year,
  dayMonth,
});

export const mockEvent: EventDetail = {
  description: null,
  title: null,
  image: null,
};

export const mockEventCompany: EventDetail = {
  ...mockEvent,
  description: 'Something happened that day for the Company',
  title: 'The Company',
  image: 'image.jpg',
};
export const mockEventDescription: EventDetail = { ...mockEvent, description: 'Something happened that day' };
export const mockEventEnlisted: EventDetail = { ...mockEvent, description: 'Main Body', title: 'Enlisted' };
export const mockEventPosted: EventDetail = { ...mockEvent, description: 'Reinforcement', title: 'Posted' };
export const mockEventTrained: EventDetail = { ...mockEvent, description: 'Military Camp, Somewhere', title: 'Trained' };
export const mockEventTitleAndDescription: EventDetail = { ...mockEvent, description: 'Something happened', title: 'Major Event' };

export const mockEventDiedOfDisease: EventDetail = { ...mockEvent, description: 'Military Hospital', title: 'Died of disease' };
export const mockEventKilledInAction: EventDetail = { ...mockEvent, description: 'Killed on the battlefield', title: 'Killed in action' };
export const mockEventDiedOfWounds: EventDetail = { ...mockEvent, description: 'Wounded on the battlefield', title: 'Died of wounds' };
export const mockEventDiedOfAccident: EventDetail = { ...mockEvent, description: 'Fatal accident', title: 'Died of accident' };
export const mockEventBuried: EventDetail = { ...mockEvent, description: 'Cemetery, France', title: 'Buried' };
export const mockEventGrave: EventDetail = { ...mockEvent, description: '1 A 26', title: 'Grave reference' };

const mockDeathEvent = (deathEvent: EventDetail): Record<string, Event[]> => ({
  1916: [
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [deathEvent, mockEventBuried, mockEventGrave],
    },
  ],
});

export const mockFrontEventsWithDiedOfDisease:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfDisease);

export const mockFrontEventsWithKilledInAction:
    Record<string, Event[]> = mockDeathEvent(mockEventKilledInAction);

export const mockFrontEventsWithDiedOfWounds:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfWounds);

export const mockFrontEventsWithDiedOfAccident:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfAccident);

export const mockFrontEvents: Record<string, Event[]> = {
  1915: [
    {
      date: mockDate('1915', '26 January'),
      event: [mockEventEnlisted],
    },
    {
      date: mockDate('1915', '30 January'),
      event: [mockEventPosted],
    },

    {
      date: mockDate('1915', '14 February'),
      event: [mockEventTrained],
    },
    {
      date: mockDate('1915', '4 May'),
      event: [mockEventTitleAndDescription],
    },
  ],
  1916: [
    {
      date: mockDate('1916', '23 June'),
      event: [mockEventDescription, mockEventDescription],
    },
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1916', '12 December'),
      event: [mockEventCompany, mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '9 April'),
      event: [mockEventTitleAndDescription, mockEventCompany],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
  ],
  1918: [
    {
      date: mockDate('1918', '3 March April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1918', '13 October'),
      event: [mockEventDescription],
    },
  ],
  1919: [
    {
      date: mockDate('1919', '4 April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1919', '12 July'),
      event: [mockEventDescription],
    },
  ],
};
