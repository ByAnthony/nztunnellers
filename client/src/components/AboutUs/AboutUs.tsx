import { Footer } from '../Footer/Footer';
import { Menu } from '../Menu/Menu';
import { Paragraph } from '../Article/Paragraph/Paragraph';
import { Title } from '../Title/Title';

import STYLES from './AboutUs.module.scss';
import { useGetAboutUsQuery } from '../../redux/slices/aboutUsSlice';

export function AboutUs() {
  const {
    data, error, isLoading, isSuccess,
  } = useGetAboutUsQuery();

  if (data) {
    return (
      <>
        {error && (
        <div className={STYLES.profile}>
          <p>An error occured</p>
        </div>
        )}
        <Menu />
        { isLoading }
        { isSuccess && (
        <div className={STYLES.container}>
          <div className={STYLES.header}>
            <Title title={data.title} />
          </div>
          <Paragraph section={data.section[0]} />
          <div className={STYLES['image-container']}>
            <img
              className={STYLES.image}
              src={`/images/about-us/${data.image[0].file}`}
              alt={data.image[0].alt}
            />
          </div>
          <Paragraph section={data.section[1]} />
        </div>
        )}
        <Footer />
      </>
    );
  }
  return null;
}
