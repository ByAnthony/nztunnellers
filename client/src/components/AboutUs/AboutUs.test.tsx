import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';

import { useGetAboutUsQuery } from '../../redux/slices/aboutUsSlice';
import { mockAboutUs } from '../../utils/mocks/mockArticle';

import { AboutUs } from './AboutUs';

jest.mock('../../redux/slices/aboutUsSlice', () => ({
  useGetAboutUsQuery: jest.fn(),
}));

test('renders profile when data is available', () => {
  (useGetAboutUsQuery as jest.Mock).mockReturnValue({
    data: mockAboutUs,
    error: null,
    isLoading: false,
    isSuccess: true,
  });

  const { asFragment } = render(<AboutUs />);

  expect(asFragment()).toMatchSnapshot();
});

test('does not render profile when data is undefined', () => {
  (useGetAboutUsQuery as jest.Mock).mockReturnValue({
    data: undefined,
    error: null,
    isLoading: false,
    isSuccess: true,
  });

  const { container } = render(<AboutUs />);

  expect(container).toBeEmptyDOMElement();
});

test('should render error page when error', () => {
  (useGetAboutUsQuery as jest.Mock).mockReturnValue({
    data: {},
    error: true,
    isLoading: false,
    isSuccess: false,
  });

  render(<AboutUs />);
  const error = screen.getByText('An error occured');

  expect(error).toBeInTheDocument();
});
