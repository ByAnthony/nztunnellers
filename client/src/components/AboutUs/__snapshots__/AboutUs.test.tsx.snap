// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders profile when data is available 1`] = `
<DocumentFragment>
  <div
    class="menu "
    data-testid="menu"
  >
    <a
      aria-label="Go to the Homepage"
      class="logo"
      href="/"
    >
      <img
        alt=""
        src="/nzt_logo.png"
      />
    </a>
    <div
      class="search-form"
    >
      <form>
        <input
          id="search"
          name="search"
          placeholder="Search for a Tunneller"
          type="text"
        />
      </form>
      <button
        aria-label="Search"
        class="search-form-button"
        type="submit"
      >
        <img
          alt=""
          height="18"
          src="/searching_hover.png"
        />
      </button>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="main-title"
      >
        <h1>
          <span
            class="title-line-1"
          >
            My Awesome
          </span>
          <span
            class="title-line-2"
          >
            Article Title
          </span>
        </h1>
      </div>
    </div>
    <div
      class="image-container"
    >
      <img
        alt="Accessible alt text"
        class="image"
        src="/images/about-us/img-123.png"
      />
    </div>
    <div>
      <div
        class="paragraph"
      >
        <h2>
          Section Title
        </h2>
        <p>
          <span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea 
          </span>
          <em>
            commodo consequat
          </em>
          <span />
          <span>
            . Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </span>
        </p>
        <p>
          <span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat
          </span>
          <span />
          <a
            href="#footnote_1"
            id="reference_1"
          >
            [1]
          </a>
          <span>
            . Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
          </span>
        </p>
        <p>
          <span>
            (6 July 2023)
          </span>
        </p>
      </div>
    </div>
    <div>
      <div
        class="paragraph"
      >
        <h2>
          Section Title
        </h2>
        <p>
          <span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor 
          </span>
          <span />
          <a
            href="http://www.loremipsum.com"
          >
            incididunt
          </a>
          <span>
             ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea 
          </span>
          <em>
            commodo consequat
          </em>
          <span />
          <span>
            . Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </span>
        </p>
        <p>
          <span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat
          </span>
          <span />
          <a
            href="#footnote_1"
            id="reference_1"
          >
            [1]
          </a>
          <span>
            . Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
          </span>
        </p>
        <p>
          <span>
            (6 July 2023)
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    class="footer"
  >
    <div
      class="map"
    >
      <div
        class="links"
      >
        <div
          class="map-link"
        >
          <a
            href="/#history"
          >
            History
          </a>
        </div>
        <div
          class="map-link"
        >
          <a
            href="/tunnellers"
          >
            Tunnellers
          </a>
        </div>
        <div
          class="map-link"
        >
          <a
            href="/about-us"
          >
            About Us
          </a>
        </div>
      </div>
      <button
        aria-label="Go back to the top of the page"
        class="scroll-top"
        type="button"
      >
        ↑
      </button>
    </div>
    <div
      class="support"
    >
      <div>
        <a
          aria-label="Go to The Artois University website"
          href="https://www.univ-artois.fr/artois-university"
        >
          <img
            alt="Artois University homepage"
            src="/images/support/logo-univ-artois-blanc_0.png"
          />
        </a>
      </div>
      <div>
        <a
          aria-label="Go to The Institute for Strategic Research website"
          href="https://www.irsem.fr/en/"
        >
          <img
            alt="Institute for Strategic Research homepage"
            src="/images/support/irsem-white.png"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
